functions.php

```diff
@@ -125,6 +125,99 @@ function custom_login_logo() {
 
 // UTILITY FUNCTIONS
 
+// Return an array of exhibition terms from these $posts.
+function get_filter_terms($posts, $type = false) {
+  if ( empty( $posts ) ) {
+    return '';
+  }
+
+  $terms = array();
+
+  foreach($posts->posts as $post) {
+
+    $post_type = get_post_type_object( $post->post_type );
+    if ($post->post_type === 'exposiciones' && $type === 'exposiciones') {
+      $post_terms = wp_get_post_terms( $post->ID, 'tipo_de_exposicion' );
+      foreach($post_terms as $post_term) {
+        echo in_array($post_term->slug, array_column($terms, 'slug'));
+        if ( in_array($post_term->slug, array_column($terms, 'slug')) == false ) {
+          $terms[] = array(
+            'slug'  =>  $post_term->slug,
+            'name'  =>  $post_term->name,
+          );
+        }
+      }
+    } else if( $post->post_type !== 'exposiciones' && !$type ) {
+      $post_type = get_post_type_object( $post->post_type );
+      if ( in_array($post_type->name, array_column($terms, 'slug')) == false ) {
+        $terms[] = array(
+          'slug'  =>  $post_type->name,
+          'name'  =>  $post_type->label,
+        );
+      }
+    } 
+  }
+  // pr( $terms );
+  return $terms;
+}
+
+// Return a list of years
+function get_all_years($post_types, $order) {
+  global $wpdb;
+
+  $where = '';
+
+  // Construct WHERE clausule for the query
+  // For all $post_types and with status 'publish'
+  // Ex. WHERE ( post_type = "post" OR post_type = "exposiciones" OR post_type = "obra" ) AND post_status = "publish "
+  for ($i = 0; $i < count($post_types); ++$i) {
+    if ( $i == 0 ) {
+      $where .= 'WHERE ( ';
+    } else {
+      $where .= ' OR ';
+    }
+    $where .= 'post_type = "' . $post_types[$i] . '"';
+  }
+  $where .= ' ) AND post_status = "publish " ';
+
+  $order = strtoupper( $order );
+  if ( $order !== 'ASC' ) {
+    $order = 'DESC';
+  }
+
+  $years = array();
+
+  // The 'last_changed' incrementor is used to invalidate the $key cache. 
+  // This way we invalidate the cache on add, delete, and update.
+  $last_changed = wp_cache_get( 'last_changed', 'posts' );
```

Comment:
ah ok i get it now. and this cache value is just for any change event on any post